src/main/java/teammates/ui/template/FeedbackResultsTable.java

```diff
@@ -24,7 +26,45 @@ public FeedbackResultsTable(int fbIndex, String studentName, FeedbackSessionResu
             for (Map.Entry<String, List<FeedbackResponseAttributes>> entry : received.entrySet()) {
                 giverIndex++;
                 this.receivedResponses.add(new FeedbackResponsePersonRow(fbIndex, giverIndex, entry.getKey(), "giver",
-                                                                         entry.getValue(), result));
+                        entry.getValue(), result));
+            }
+
+            boolean hasAnsweredContrib = false;
+            boolean contribExistsFromOtherStudent = false;
+            FeedbackResponseAttributes response = new FeedbackResponseAttributes();
+
+            for (Map.Entry<String, List<FeedbackResponseAttributes>> entry : received.entrySet()) {
+                for (int j = 0; j < entry.getValue().size(); j++) {
+                    FeedbackQuestionType questionType = entry.getValue().get(j).feedbackQuestionType;
+                    String recipient = entry.getValue().get(j).recipient;
+                    String giver = entry.getValue().get(j).giver;
+                    if (questionType.equals(FeedbackQuestionType.CONTRIB)) {
+                        response = new FeedbackResponseAttributes(entry.getValue().get(j));
+                        contribExistsFromOtherStudent = true;
+                    }
+                    if (giver.equals(recipient) && questionType.equals(FeedbackQuestionType.CONTRIB)) {
+                        hasAnsweredContrib = true;
+                    }
+                }
+            }
+            /*
+             * if student has not submitted the session or CONTRIB response
+             * about himself then student's submission to CONTRIB is shown No
+             * Response and PC is shown next to it
+             */
+            if ((!submissionStatus || !hasAnsweredContrib) && contribExistsFromOtherStudent) {
+                giverIndex++;
+                response.giver = response.recipient;
+                response.giverSection = response.recipientSection;
+                List<FeedbackResponseAttributes> responses = new ArrayList<>();
+                responses.add(response);
+                FeedbackResponsePersonRow contribPcRow = new FeedbackResponsePersonRow(fbIndex, giverIndex, studentName,
+                                                                                        "giver", responses, result);
+                String responseText = contribPcRow.getResponses().get(0).getResponseText();
```

Comment:
Is `contribPcRow.getResponses()` guaranteed to be non-empty?

Reply:
Yes. this block of code executes only if `contribPcRow.getResponses()` is non-empty as made sure by the `if(..)` condition.

Reply:
`if ((!submissionStatus || !hasAnsweredContrib) && contribExistsFromOtherStudent)`